src/lib/storage/storage_manager.hpp

```diff
@@ -43,7 +43,7 @@ class StorageManager : private Noncopyable {
  protected:
   StorageManager() {}
   StorageManager& operator=(StorageManager&&) = default;
-
-  // Implementation goes here
+  std::map<std::string, std::shared_ptr<Table>> tables;
+  static StorageManager* instance;
```

Comment:
Generally, use smart pointers rather than raw pointers. However, even a smart pointer member is not necessary here since `instance` is never used but the static local `static StorageManager sm` is used instead.